platform/openide.filesystems/src/org/openide/filesystems/Ordering.java

```diff
@@ -158,7 +164,8 @@ public int compareTo(ChildAndPosition o) {
                 previousChild = subsequentChild;
             }
         }
-        if (logWarnings && /* #201893*/ !parent.getPath().matches("Projects/.+/Lookup") && !childrenByPosition.isEmpty() && childrenByPosition.size() < children.size()) {
+        if (logWarnings && /* #201893*/ !parent.getPath().matches("Projects/.+/Lookup") && !childrenByPosition.isEmpty() && childrenByPosition.size() < children.size() && 
```

Comment:
Couldn't, by a chance, the check for `Projects/.+/Lookup` be removed by using the same trick with autogenerating the position? No need to solve it in this PR however.